pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/QueryOptimizer.java

```diff
@@ -31,12 +31,14 @@
 import org.apache.pinot.core.query.optimizer.filter.FlattenAndOrFilterOptimizer;
 import org.apache.pinot.core.query.optimizer.filter.MergeEqInFilterOptimizer;
 import org.apache.pinot.core.query.optimizer.filter.MergeRangeFilterOptimizer;
+import org.apache.pinot.core.query.optimizer.filter.NumericalFilterOptimizer;
 import org.apache.pinot.spi.data.Schema;
 
 
 public class QueryOptimizer {
   private static final List<FilterOptimizer> FILTER_OPTIMIZERS = Arrays
-      .asList(new FlattenAndOrFilterOptimizer(), new MergeEqInFilterOptimizer(), new MergeRangeFilterOptimizer());
+      .asList(new FlattenAndOrFilterOptimizer(), new MergeEqInFilterOptimizer(), new MergeRangeFilterOptimizer(),
+          new NumericalFilterOptimizer());
```

Comment:
`NumericalFilterOptimizer` should be inserted before the `MergeEqInFilterOptimizer` so that `MergeEqInFilterOptimizer` and `MergeRangeFilterOptimizer` can read the optimized values

Reply:
Ordering doesn't seem to be very important here with respect to performance since we are working on small constant size operator trees. Seems to me that it would be better to keep these rewrite classes order independent, so that new rewrites can be easily plugged in and obsolete once can be deleted without worrying out ordering dependencies. But let me know if you still prefer moving it. The only thing that may cause rewrite issues would probably be IN clause with large number of values and in that case it would make sense to traverse IN clause only once.

Reply:
It is not about performance, but correctness. Currently `MergeRangeFilterOptimizer` assumes the numeric format is correct. If we put `NumericalFilterOptimizer` before `MergeRangeFilterOptimizer`, `int_col > 1.5 AND int_col < 2.5` will work.

Reply:
I see what you mean. Can we look at this with next PR when we add type casting of range operators? This PR is only doing = and != operator, so even if the class is moved, `int_col > 1.5 AND int_col < 2.5` still wouldn't work.

Reply:
We will very likely forget about this if it is not added in the correct order in the first PR. Let's put `NumericalFilterOptimizer` after `FlattenAndOrFilterOptimizer` in this PR because that is the correct order in the final state. All the optimizers based on filter values should follow this optimizer.

Reply:
Moved.